feathr_project/feathr/utils/_envvariableutil.py

```diff
@@ -24,51 +29,59 @@ def get_environment_variable_with_default(self, *args):
+                    return yaml_layer
+                except KeyError as exc:
+                    logger.warning("{} not found in the config file, loading it in key vault.", env_keyword)
+                    # if not found in the config file, use key vault to get the value
+                    # return self.akv_client.get_akv_secret(env_keyword)  if self.akv_name else ""
+                except yaml.YAMLError as exc:
+                    logger.warning(exc)
+        # If it's not available in the feathr_config.yaml file, Feathr will try to reterive the value from key vault
+        if self.akv_name:
             try:
-                yaml_config = yaml.safe_load(stream)
-                # concat all layers
-                # check in environment variable
-                yaml_layer = yaml_config
-
-                # resolve one layer after another
-                for arg in args:
-                    yaml_layer = yaml_layer[arg]
-                return yaml_layer
-            except KeyError as exc:
-                logger.info(exc)
+                return  self.akv_client.get_feathr_akv_secret(env_keyword) 
+            except ResourceNotFoundError:
+                # print out warning message if cannot find the env variable in all the resources
+                logger.warning('Environment variable {} not found in environment variable, default YAML config file, or key vault service.', env_keyword)
                 return ""
-            except yaml.YAMLError as exc:
-                logger.info(exc)
 
-    @staticmethod
-    def get_environment_variable(variable_key):
+    def get_environment_variable(self, variable_key):
         """Gets the environment variable for the variable key.
+        
 
         Args:
             variable_key: environment variable key that is used to retrieve the environment variable
         Return:
-            A environment variable for the variable key.
+            A environment variable for the variable key. It will retrieve the value of the environment variables in the following order:
+            If the key is set in the envrionment variable, Feathr will use the value of that environment variable
+            If it's not available in the environment variable file, Feathr will try to reterive the value from key vault
         Raises:
             ValueError: If the environment variable is not set for this key, an exception is thrown.
             """
         password = os.environ.get(variable_key)
-        if not password:
-            logger.info(variable_key +
-                        ' is not set in the environment variables.')
-            akv_name = os.environ.get("KEY_VAULT_NAME")
-            if akv_name:
-                logger.info('Fetching the value {} from Key Vault.', variable_key)
-                akv_client = AzureKeyVaultClient(akv_name)
-                password = akv_client.get_akv_secret(variable_key)
-        return password
 
+        if password:
+            return password
```

Comment:
>password [](http://example.com/codeflow?start=19&length=8)

get_environment_variable is a generic method and use password as value name here is confused.

Reply:
renamed to `env_var_value`